### PR TITLE
Added tcgetpgrp and tcsetpgrp

### PR DESCRIPTION
**Pull request by zethra**
I added the tcgetpgrp and tcsetpgrp functions.  I'm not sure if I did everything right so please tell me if I need to change anything.

### CODE REVIEW

**Comment by posborne** [comment]
Thank you!  The implementation looks good to me.  Can you do two minor things:

1. Add documentation to the functions following the basic style present in the changes from this PR (which should be merged soon): https://github.com/nix-rust/nix/pull/449
2. Please add a note about the new functions to the CHANGELOG.md file in the root of the project.

**Comment by zethra** [comment]
Sure. I'll get in that.

File: src/unistd.rs [hunk 1]
```diff
@@ -68,6 +68,17 @@ pub fn getpgid(pid: Option<pid_t>) -> Result<pid_t> {
...
 
+#[inline]
+pub fn tcgetgrp(fd: c_int) -> Result<pid_t> {
+    let res = unsafe { libc::tcgetpgrp(fd) };
+    Errno::result(res)
+}
+#[inline]
+pub fn tcsetpgrp(fd: c_int, pgrp: pid_t) -> Result<c_int> {
+    let res = unsafe { libc::tcsetpgrp(fd, pgrp) };
+    Errno::result(res)
```
Comment (kamalmarhubi) [inline src/unistd.rs]: this should have `.map(drop)`, as the return only distinguishes failures.

**Comment by zethra** [comment]
I've made the request changes.

**Comment by kamalmarhubi** [comment]
Thanks @zethra!

@homu r+

**Comment by kamalmarhubi** [comment]
@homu retry

**Pull request merged**